drivers/timer/cortex_m_systick.c

```diff
@@ -87,15 +87,17 @@ void z_clock_isr(void *arg)
 {
 	ARG_UNUSED(arg);
 	u32_t dticks;
+	u32_t key = irq_lock();
 
-	cycle_count += last_load;
+	elapsed();
+	cycle_count += overflow_cyc;
+	overflow_cyc = 0;
 	dticks = (cycle_count - announced_cycles) / CYC_PER_TICK;
 	announced_cycles += dticks * CYC_PER_TICK;
 
-	overflow_cyc = SysTick->CTRL; /* Reset overflow flag */
-	overflow_cyc = 0U;
-
 	z_clock_announce(TICKLESS ? dticks : 1);
+
+	irq_unlock(key);
```

Comment:
I'm very certain you do **not** want the lock held during `z_clock_announce()` which can take arbitrarily long.   Move it before that call, please.

Reply:
@pabigot thanks! pushed again.